src/__tests__/storybook.test.tsx

```diff
@@ -0,0 +1,25 @@
+import fs from 'fs'
+import glob from 'fast-glob'
+
+// Components opted into the new story format
+const components = ['TreeView']
+
+const storyFiles = glob
+  .sync('src/**/*.stories.tsx')
+  .filter(file => components.some(component => file.includes(component)))
+  .filter(file => !file.endsWith('.features.stories.tsx'))
+
+for (const storyFile of storyFiles) {
+  describe(`${storyFile}`, () => {
+    const storyFileContents = fs.readFileSync(storyFile, 'utf8')
+
+    test('exports a Default story', () => {
+      expect(storyFileContents).toMatch(/export const Default/)
```

Comment:
Is there a more robust way to check the exports of the story file? Maybe with ts-morph? cc @joshblack @siddharthkp

Reply:
@colebemis through up a draft here of a possible approach: https://github.com/primer/react/pull/2537

Curious what you think!